### PR TITLE
Ensure _blockHoist is set on function destructuring defaults

### PR DESCRIPTION
**Pull request by loganfsmyth**
Fixes https://github.com/babel/babel/issues/1908.

### CODE REVIEW

File: src/babel/transformation/transformers/es6/destructuring.js [hunk 1]
```diff
@@ -316,9 +316,12 @@ class DestructuringTransformer {
 
     var left = pattern.left;
     if (t.isPattern(left)) {
-      this.nodes.push(t.expressionStatement(
+      let tempValueDefault = t.expressionStatement(
```
Comment (sebmck) [inline src/babel/transformation/transformers/es6/destructuring.js]: Use `var` please.

Comment (loganfsmyth) [inline src/babel/transformation/transformers/es6/destructuring.js]: Fixed.

**Comment by sebmck** [comment]
Squash and this is GTM. Thanks!

**Comment by sebmck** [comment]
Thanks @loganfsmyth!

**Pull request merged**